wgpu-core/src/device/mod.rs

```diff
@@ -981,17 +981,17 @@ impl<A: HalApi> Device<A> {
 
             let mut required_features = wgt::Features::empty();
             let mut required_downlevel_flags = wgt::DownlevelFlags::empty();
-            let (array_feature, is_writable_storage) = match entry.ty {
+            let (array_feature, storage_buffer, is_writable_storage) = match entry.ty {
                 Bt::Buffer {
                     ty: wgt::BufferBindingType::Uniform,
                     has_dynamic_offset: false,
                     min_binding_size: _,
-                } => (Some(wgt::Features::BUFFER_BINDING_ARRAY), false),
+                } => (Some(wgt::Features::BUFFER_BINDING_ARRAY), false, false),
```

Comment:
Would you mind turning this into an enum? booleans in such position are bad, and having two of them is twice as bad:)